pkg/controller/blockdeviceclaim/blockdeviceclaim_controller_test.go

```diff
@@ -238,6 +254,100 @@ func (r *ReconcileBlockDeviceClaim) InvalidCapacityTest(t *testing.T,
 	r.CheckBlockDeviceClaimStatus(t, req, openebsv1alpha1.BlockDeviceClaimStatusInvalidCapacity)
 }
 
+func TestBlockDeviceClaimsLabelSelector(t *testing.T) {
+	// Set the logger to development mode for verbose logs.
+	logf.SetLogger(logf.ZapLogger(true))
+
+	// Create a fake client to mock API calls.
+	cl, s := CreateFakeClient()
+
+	// Create a ReconcileDevice object with the scheme and fake client.
+	r := &ReconcileBlockDeviceClaim{client: cl, scheme: s}
+
+	// Mock request to simulate Reconcile() being called on an event for a
+	// watched resource .
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      blockDeviceClaimName,
+			Namespace: namespace,
+		},
+	}
+
+	tests := map[string]struct {
+		bdLabels           map[string]string
+		selector           *metav1.LabelSelector
+		expectedClaimPhase openebsv1alpha1.DeviceClaimPhase
+	}{
+		"only hostname label is present and no selector": {
+			bdLabels: map[string]string{
+				kubernetes.KubernetesHostNameLabel: fakeHostName,
+			},
+			selector:           nil,
+			expectedClaimPhase: openebsv1alpha1.BlockDeviceClaimStatusDone,
+		},
+		"no label is present and no selector": {
+			bdLabels:           map[string]string{},
+			selector:           nil,
+			expectedClaimPhase: openebsv1alpha1.BlockDeviceClaimStatusDone,
+		},
+		"conflicting label selector and hostname": {
+			bdLabels: map[string]string{},
+			selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					kubernetes.KubernetesHostNameLabel: "hostname1",
+				},
+			},
+			expectedClaimPhase: openebsv1alpha1.BlockDeviceClaimStatusPending,
+		},
+		"custom labels on bd": {
+			bdLabels: map[string]string{
+				"ndm.io/test": "1234",
+			},
+			selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"ndm.io/test": "1234",
+				},
+			},
+			expectedClaimPhase: openebsv1alpha1.BlockDeviceClaimStatusDone,
+		},
```

Comment:
Adding one test case to cover - custom label and hostname will be good.

Reply:
Done.